asciidoc/courses/llm-fundamentals/modules/1-introduction/lessons/3-grounding/lesson.adoc

```diff
@@ -0,0 +1,89 @@
+= Grounding LLMs
+:order: 3
+
+// TODO: Do we include in-context learning, fine-tuning or prompt emgineering?
+
+In the previous lesson, you learned about the potential methods that you could use to avoid Hallucination.
+The last point mentioned was **Grounding**.
+
+For developers and data scientists, this method offers the highest impact for the lowest effort.
+
+In the previous lesson, we mentioned an example chatbot for a news agency.
+Let's drill down into that example.
+
+== Data Cut-Off Dates
+
+As you can imagine, training a Large Language Model has a high computational cost.
+According to Wikipedia, OpenAI's GPT-3 model was link:https://en.wikipedia.org/wiki/GPT-3[trained on 175 billion parameters^], and the resulting trained model takes 800GB to store.
+
+This would be expensive and time-consuming to constantly train on real data.
+A model the size of GPT-3 would typically take weeks to months to train on supercomputing clusters with multiple GPUs or TPUs.
+
+For a news agency providing real-time updates on fast-moving breaking news would be out of the question.
+
+In this case, it makes sense to train a model on static data and supplement the pre-existing knowledge base with real-time data retrieved from up-to-date sources.
+
+This is where **Retrieval Augmented Generation**, or **RAG** comes in.
+
+
+== Retrieval Augmented Generation
+
+Retrieval Augmented Generation (RAG) is a hybrid approach that combines the strengths of large-scale language models with external retrieval or search mechanisms, enabling the model to dynamically pull relevant information from vast datasets during the generation process, thereby enhancing its ability to provide detailed and contextually accurate responses.
```

Comment:
the model doesn't pull data from the datasets it's the application that does it and feed it to the model.